src/hotspot/share/compiler/compilerDirectives.cpp

```diff
@@ -621,7 +619,7 @@ DirectiveSet* DirectiveSet::clone(DirectiveSet const* src) {
 #undef copy_string_members_definition
 
   set->_intrinsic_control_words = src->_intrinsic_control_words;
-  set->_ideal_phase_name_mask = src->_ideal_phase_name_mask;
+  set->_ideal_phase_name_set.set_from(src->_ideal_phase_name_set);
```

Comment:
The advantage of passing the set/mask directly is that here you could also use your `set_ideal_phase_name_set` method.

Reply:
Thanks, updated.